braids/settings.h

```diff
@@ -323,7 +323,7 @@ class Settings {
   }
   
   inline int32_t pitch_transposition() const {
-    int32_t t = data_.pitch_range == PITCH_RANGE_LFO ? -36 << 7 : 0;
+    int32_t t = data_.pitch_range == PITCH_RANGE_LFO ? (unsigned)-36 << 7 : 0;
```

Comment:
Hmm, this seems a bit strange.
While the commit silences the warning, it might also alter the outcome to a positive value.
- if it is intended to be positive, why not remove the sign instead of `(unsigned)-`?

```c
[MacPro]martin:~/tmp/c$ cat test.c
#include <stdio.h>
int main() {
	int32_t i1 = -32 << 7;
	int32_t i2 = (unsigned)32 << 7;
	int32_t i3 = 32 << 7;
	printf("%d %d %d\n", i1, i2, i3);
}
```

```
[MacPro]martin:~/tmp/c$ gcc test.c
test.c:5:19: warning: shifting a negative signed value is undefined
      [-Wshift-negative-value]
        int32_t i1 = -32 << 7;
                     ~~~ ^
1 warning generated.
```

```
[MacPro]martin:~/tmp/c$ ./a.out
-4096 4096 4096
```

Reply:
-(36 << 7) then?

Reply:
yeah putting parenthesis seems to be the right choice if negative values are the intention.
I was not sure if that was the case or not, often times it is not.
Sorry for that, should have asked.

Reply:
also your test case is missing the actual change of `(unsigned)-32 << 7;`

Reply:
just tested here. `(unsigned)-32 << 7;` `-32 << 7;` and `-(32 << 7);` all print the same integer value
so nothing really broke with the PR

your suggestion of `-(36 << 7)` seems the best one though, a it makes clear what is the intention in the code.

Reply:
> also your test case is missing the actual change of `(unsigned)-32 << 7;`

Ah, sorry, you're right.
Yep, @pichenettes suggestion seems clearer nevertheless.